src/utils/convert3Dto4D.m

```diff
@@ -0,0 +1,200 @@
+  optSource.sequenceList = {sourceDataStruc(isDir).name}';
+
+  % Loop through the sequence folders
+
+  tic;
+
+  for iSeq = 1:size(optSource.sequenceList, 1)
+
+    % Skip 'non' folders
+    if length(char(optSource.sequenceList(iSeq))) > 2
+
+      % Check if sequence to ignore or not
+      if contains(optSource.sequenceList(iSeq), optSource.sequenceToIgnore)
+
+        newline;
+        warning('IGNORING SEQUENCE: %s', string(optSource.sequenceList(iSeq)));
+        newline;
+
+      else
+
+        fprintf('\n\nCONVERTING SEQUENCE: %s \n\n', char(optSource.sequenceList(iSeq)));
+
+        % Set whether to remove dummies or not
+
+        nbDummies = 0;
+
+        if contains(optSource.sequenceList(iSeq), optSource.sequenceRmDummies)
+
+          nbDummies = optSource.nbDummies;
+
+          fprintf('\n\nREMOVING %s DUMMIES\n\n', num2str(nbDummies));
+
+        end
+
+        % Get sequence folder path
+        sequencePath = char(fullfile(optSource.sourceDir, optSource.sequenceList(iSeq)));
+
+        % Retrieve volume files info
+        [volumesList, outputNameImage] = parseFiles('nii', sequencePath, nbDummies);
+
+        % Set output name, it takes the file name of the 1st volume of the 4D file and add subfix
+        outputNameImage = strrep(outputNameImage, '.nii', '_4D.nii');
+
+        % Retrieve sidecar json files info
+        [jsonList, outputNameJson] = parseFiles('json', sequencePath, nbDummies);
+
+        jsonFile = spm_jsonread(char(jsonList(1)));
+
+        % % % % % % LIEGE  SPECIFIC % % % % % % %
+        RT = jsonFile.acqpar.RepetitionTime / 1000;
+        % % % % % % % % % % % % % % % % % % % % %
+
+        % Set and run spm batch, input all the volumes minus the dummies if > 0
+        matlabbatch = setBatch3Dto4D(volumesList(nbDummies + 1:end, :), ...
+                                     outputNameImage, ...
+                                     optSource.dataType, ...
+                                     RT);
+
+        spm_jobman('run', matlabbatch);
+
+        if optSource.zip
+
+          % Zip and delete the and the new 4D file
+          fprintf(1, 'ZIP AND DELETE THE NEW 4D BRAIN \n\n');
+
+          gzip([sequencePath filesep outputNameImage]);
+
+          delete([sequencePath filesep outputNameImage]);
+
+        end
+
+        % Save one sidecar json file, it takes the file name of the 1st volume of the 4D file and
+        % add subfix
+        if ~isempty(jsonList)
+
+          copyfile(char(jsonList(1)), [sequencePath filesep strrep(outputNameJson, '.json', '_4D.json')]);
+
+        end
+
+        % Delete all the single volumes .nii and .json files
+        fprintf(1, 'EXTERMINATE SINGLE VOLUMES FILES \n\n');
+
+        for iDel = 1:length(volumesList)
+
+          delete(char(volumesList(iDel)));
+          delete(char(jsonList(iDel)));
+
+        end
+
+      end
+
+    end
+
+  end
+
+  newline;
+  toc;
+
+end
+
+function [fileList, outputName] = parseFiles(fileExstention, sequencePath, nbDummies)
```

Comment:
```suggestion
function [fileList, outputName] = parseFiles(fileExtention, sequencePath, nbDummies)
```

Just a typo in the variable name.

Will let you fix in the rest of this function. :-)